typescript/cli/src/config/warp.ts

```diff
@@ -40,75 +60,68 @@ export async function createWarpRouteDeployConfig({
   );
 
-  const isNative = await confirm({
-    message:
-      'Are you creating a route for the native token of the base chain (e.g. Ether on Ethereum)?',
-  });
+  const result: WarpRouteDeployConfig = {};
+  for (const chain of warpChains) {
+    logBlue(`Configuring warp route for chain ${chain}`);
+    const type = await select({
+      message: `Select ${chain}'s token type`,
+      choices: TYPE_CHOICES,
+    });
 
-  const isNft = isNative
-    ? false
-    : await confirm({ message: 'Is this an NFT (i.e. ERC-721)?' });
-  const isYieldBearing =
-    isNative || isNft
-      ? false
-      : await confirm({
-          message:
-            'Do you want this warp route to be yield-bearing (i.e. deposits into ERC-4626 vault)?',
-        });
+    // TODO: restore NFT prompting
+    const isNft =
+      type === TokenType.syntheticUri || type === TokenType.collateralUri;
 
-  const addressMessage = `Enter the ${
-    isYieldBearing ? 'ERC-4626 vault' : 'collateral token'
-  } address`;
-  const baseAddress = isNative
-    ? ethers.constants.AddressZero
-    : await input({ message: addressMessage });
+    // TODO: migrate to detectAndConfirmOrPrompt
+    const addresses = await context.registry.getChainAddresses(chain);
+    const mailbox =
+      addresses?.mailbox ??
+      (await input({
+        message: `Enter the mailbox address for chain ${chain}`,
+      }));
 
-  const metadataWithoutBase = objFilter(
-    context.chainMetadata,
-    (chain, _): _ is ChainMetadata => chain !== baseChain,
-  );
-  const syntheticChains = await runMultiChainSelectionStep(
-    metadataWithoutBase,
-    'Select chains to which the base token will be connected',
-  );
-
-  // TODO add more prompts here to support customizing the token metadata
-  let result: WarpRouteDeployConfig;
-  if (isNative) {
-    result = {
-      [baseChain]: {
-        type: TokenType.native,
-      },
-    };
-  } else {
-    result = {
-      [baseChain]: {
-        type: isYieldBearing ? TokenType.collateralVault : TokenType.collateral,
-        token: baseAddress,
-        isNft,
-      },
-    };
+    switch (type) {
+      case TokenType.collateral:
+      case TokenType.collateralXERC20:
+      case TokenType.collateralFiat:
+      case TokenType.collateralUri:
+      case TokenType.fastCollateral:
+      case TokenType.collateralVault:
+        result[chain] = {
+          mailbox,
+          type,
+          owner,
+          isNft,
+          token: await input({
+            message: `Enter the existing token address for chain ${chain}`,
+          }),
+        };
+        break;
+      default:
+        result[chain] = { mailbox, type, owner, isNft };
+    }
   }
 
-  syntheticChains.map((chain) => {
-    result[chain] = {
-      type: TokenType.synthetic,
-    };
-  });
-
   if (isValidWarpRouteDeployConfig(result)) {
     logGreen(`Warp Route config is valid, writing to file ${outPath}`);
     writeYamlOrJson(outPath, result);
   } else {
     errorRed(
```

Comment:
is this error useful? isn't this function going to create the yaml?

if the function fails to create the yaml, how would i compare it to the example?

if we remove this error, then this conditional can be removed, and we can just `.parse`, and then `writeYamlOrJson`